dali/python/nvidia/dali/plugin/mxnet.py

```diff
@@ -130,37 +141,56 @@ def __next__(self):
         for p in self._pipes:
             outputs.append(p._share_outputs())
         for i in range(self._num_gpus):
-            out_data = []
-            out_label = []
             # MXNet wants batches with clear distinction between
             # data and label entries, so segregate outputs into
             # 2 categories
+            category_outputs = {key : [] for key in self._output_categories}
             for j, out in enumerate(outputs[i]):
-                if self.output_map[j] == "data":
-                    out_data.append(out)
-                elif self.output_map[j] == "label":
-                    out_label.append(out)
-
+                category_outputs[self._output_categories_map[j]].append(out)
             # Change DALI TensorLists into Tensors
-            data = list(map(lambda x: x.as_tensor(), out_data))
-            data_info = list(map(lambda x: (x.shape(), np.dtype(x.dtype())), data))
-            label = list(map(lambda x: x.as_tensor(), out_label))
-            # Change label shape from [batch_size, 1] to [batch_size]
-            for l in label:
-                l.squeeze()
-            label_info = list(map(lambda x: (x.shape(), np.dtype(x.dtype())), label))
+            category_tensors = dict()
+            category_info = dict()
+            # For data proceed normally
+            category_tensors[DALIGenericIterator.DATA_TAG] = \
+                [x.as_tensor() for x in category_outputs[DALIGenericIterator.DATA_TAG]]
+            category_info[DALIGenericIterator.DATA_TAG] = \
+                [(x.shape(), np.dtype(x.dtype())) for x in category_tensors[DALIGenericIterator.DATA_TAG]]
+            # For labels we squeeze the tensors
+            category_tensors[DALIGenericIterator.LABEL_TAG] = \
+                [x.as_tensor() for x in category_outputs[DALIGenericIterator.LABEL_TAG]]
+            for label in category_tensors[DALIGenericIterator.LABEL_TAG]:
+                label.squeeze()
+            category_info[DALIGenericIterator.LABEL_TAG] = \
+                [(x.shape(), np.dtype(x.dtype())) for x in category_tensors[DALIGenericIterator.LABEL_TAG]]
+
             # If we did not yet allocate memory for that batch, do it now
             if self._data_batches[i][self._current_data_batch] is None:
-                d = [mx.nd.zeros(shape, mx.gpu(self._pipes[i].device_id), dtype = dtype) for shape, dtype in data_info]
-                l = [mx.nd.zeros(shape, mx.cpu(0), dtype = dtype) for shape, dtype in label_info]
+                mx_gpu_device = mx.gpu(self._pipes[i].device_id)
+                mx_cpu_device = mx.cpu(0)
+                from nvidia.dali.backend import TensorGPU
```

Comment:
Why here?

Reply:
It was the same in pytorch plugin, to limit the scope of the import.